app/models/user.rb

```diff
@@ -19,30 +19,59 @@ class User < ApplicationRecord
   has_many :likes, dependent: :destroy
   has_many :liked_posts, through: :likes, source: 'post'
   has_many :friendships
+  has_many :received_but_unconfirmed, -> { where confirmed: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :pending_friends, through: :received_but_unconfirmed, source: 'user'
   has_many :inverted_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
   scope :all_except, ->(me) { where.not(id: me) }
+  has_many :confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship'
+  has_many :friends, through: :confirmed_friendships, source: 'user'
```

Comment:
Currently, this just returns the friends for one of the users involved in the friendship. Please, follow the suggestion I left you in the previous comment. With that approach, you won't need additional methods to get user friends.

Reply:
# Guys, I have separated each required feature test so it can be clearer for the next person who reviews this project.
I have take notice of some the suggestions made by @EFRAIN and I would make that implementation in m9 when I refactor the whole app